src/Symfony/Component/Dotenv/Command/DebugCommand.php

```diff
@@ -36,8 +36,8 @@ final class DebugCommand extends Command
      */
     protected static $defaultDescription = 'Lists all dotenv files with variables and values';
 
-    private $kernelEnvironment;
-    private $projectDirectory;
+    private readonly string $kernelEnvironment;
+    private readonly string $projectDirectory;
```

Comment:
Adding `readonly` provides basically nothing when used on private properties
I'd be in favor of not using it on existing code to reduce merge conflicts.

Reply:
In this particular case, we're also adding a type which will cause conflicts anyway. We could btw migrate directly to CPP.

Reply:
> I agree on using CPP, the question is: Could a minor version introduce this type of API changes?

CPP alone does not change the API of a class.

Reply:
> I might be wrong but could you explain which cases can trigger a merge conflict when using `readonly` attributes?

We continuously merge lower maintained branches into higher ones. Changes that are purely cosmetic might make those merges more difficult and error-prone.

Reply:
> I mean, except maybe when `7.0` will be merged but the process will be the same if I'm right 😅

We usually maintain branches of two or three major releases at the same time. At the moment, we merge `4.4` into `5.4` into `6.0` into `6.1.` into `6.2`.

Reply:
CPP introduced and `readonly` attributes removed 🙂